src/test/java/com/ibm/csync/SubTestsIT.java

```diff
@@ -38,13 +38,16 @@
 
     @Before
     public void setup() throws Exception{
+        CompletableFuture<Boolean> future = new CompletableFuture<>();
         int x = 50;
         csync = CSync.builder()
-                .token(System.getenv("CSYNC_DEMO_TOKEN"))
-                .provider(System.getenv("CSYNC_DEMO_PROVIDER"))
                 .host(System.getenv("CSYNC_HOST"))
                 .port(Integer.parseInt(System.getenv("CSYNC_PORT")))
                 .build();
+        csync.authenticate(System.getenv("CSYNC_DEMO_PROVIDER"), System.getenv("CSYNC_DEMO_TOKEN"))
+                .onComplete((ex, isSuccessful) -> future.complete(true));
+
+        assertTrue(future.get(10, TimeUnit.SECONDS));
```

Comment:
This is weird. We don't need to assertTrue here because its in the setup and it will automatically assume true if we get passed the future. We should however, fail("Unable to login with provided CSYNC_DEMO_PROVIDER and CSYNC_DEMO_TOKEN environmental variables") if the future comes back with an error. This is the same for the rest of the areas you do this